checker/src/main/java/org/checkerframework/checker/determinism/qual/RequiresDetToString.java

```diff
@@ -0,0 +1,14 @@
+package org.checkerframework.checker.determinism.qual;
+
+import java.lang.annotation.*;
+import org.checkerframework.framework.qual.InheritedAnnotation;
+
+/**
+ * A method with this annotation requires that all arguments corresponding to {@code Object}
```

Comment:
The comment explains the what the checker enforces when this annotation is on a method which is good.  But there should also be some explanation about the meaning of this annotation.  Something like, "Method with this annotation expect a deterministic result when calling toString on Object arguments."